### PR TITLE
images: module cleanup

### PR DESCRIPTION
**Pull request by ibgreen**
Some good changes were made in the PR that attempted to split the images module. Even though that PR (https://github.com/visgl/loaders.gl/pull/746) was closed, it makes sense to take in these changes.

- Before `modules/images/dist/dist.min.js 12K`
- After `modules/images/dist/dist.min.js 11K`

### CODE REVIEW

File: modules/images/docs/api-reference/binary-image-api.md [hunk 1]
```diff
@@ -17,52 +17,27 @@ The format is reported using MIME types strings. Supported binary formats and th
 const response = await fetchFile(imageUrl);
 const arrayBuffer = await response.arrayBuffer();
 
-const mimeType = getBinaryImageMIMEType(arrayBuffer);
-const {width, height} = getBinaryImageSize(arrayBuffer, mimeType);
+const metadata = getBinaryImageMetadata(arrayBuffer);
+if (medata) {
+  const {width, height, mimeTupe} = metadata;
```
Comment (xintongxia) [inline modules/images/docs/api-reference/binary-image-api.md:22]: `mimeType`

File: modules/images/docs/api-reference/binary-image-api.md [hunk 2]
```diff
@@ -17,52 +17,27 @@ The format is reported using MIME types strings. Supported binary formats and th
...
-Returns:
-
-- an object with fields containing the size of the image represented by the data.
+Returns a metadata object with if the binary data represents a known binary image format.
 
 ```js
 {
-  width: Number,
-  height: Number
+  metadata: string;
```
Comment (xintongxia) [inline modules/images/docs/api-reference/binary-image-api.md:38]: metadata is an object.

File: test/modules.js [hunk 3]
```diff
@@ -6,47 +6,72 @@ const {_addAliases} = require('@loaders.gl/loader-utils');
...
 _addAliases(ALIASES);
 
+const TEST_CORE = false;
+const TEST_IMAGES = true;
+const TEST_MESHES = false;
+const TEST_SCENEGRAPHS = false;
+const TEST_TILES = false;
+const TEST_GEOSPATIAL = false;
+const TEST_TABLES = false;
+const TEST_ARCHIVES = false;
```
Comment (xintongxia) [inline test/modules.js:16]: Why are only image tests enabled?

File: modules/images/docs/api-reference/binary-image-api.md [hunk 4]
```diff
@@ -17,52 +17,27 @@ The format is reported using MIME types strings. Supported binary formats and th
...
-
-Parameters:
-
-- `imageData`: Binary encoded image data.
-- `mimeType`: A MIME type string
-
-Returns:
-
-- an object with fields containing the size of the image represented by the data.
+Returns a metadata object with if the binary data represents a known binary image format.
```
Comment (kylebarron) [inline modules/images/docs/api-reference/binary-image-api.md:34]: `with if`

File: modules/images/docs/api-reference/binary-image-api.md [hunk 5]
```diff
@@ -17,52 +17,27 @@ The format is reported using MIME types strings. Supported binary formats and th
...
-
-Parameters:
-
-- `imageData`: Binary encoded image data.
-- `mimeType`: A MIME type string
-
-Returns:
-
-- an object with fields containing the size of the image represented by the data.
+Returns a metadata object describing the image. Returns `null` if the binary data does not represent a known binary image format.
```
Comment (Pessimistress) [inline modules/images/docs/api-reference/binary-image-api.md:34]: This conflicts with the statement below (throws if image is not in a supported binary format)

File: modules/images/src/index.js [hunk 6]
```diff
@@ -3,33 +3,36 @@ export {default as ImageWriter} from './image-writer';
...
 export {default as HTMLImageLoader} from './image-loader';
 
-import {getDefaultImageType} from './lib/parsed-image-api/image-type';
+import {getDefaultImageType} from './lib/category-api/image-type';
 
 export function getSupportedImageType(imageType = null) {
   return getDefaultImageType();
 }
+
+// Binary Image API
```
Comment (Pessimistress) [inline modules/images/src/index.js:33]: Add `TODO - remove in v3`

**Pull request merged**